src/main/java/com/datalogics/pdf/hsm/samples/LunaHsmManager.java

```diff
@@ -0,0 +1,142 @@
+        } else {
+            slotManager.login(tokenLabel, password);
+        }
+
+        return slotManager.isLoggedIn();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.datalogics.pdf.hsm.samples.HsmManager#hsmLogout()
+     */
+    @Override
+    public void hsmLogout() {
+        /*
+         * When you are done using the Luna HSM, it is customary to log out of the HSM to prevent unauthorized access at
+         * a later point in your application.
+         *
+         * Only use the LunaSlotManager.logout() method if you used one of the LunaSlotManager.login() methods for
+         * opening access to the HSM. If you use an external login method, you will need to use an external logout
+         * method.
+         */
+        slotManager.logout();
+    }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.datalogics.pdf.hsm.samples.HsmManager#getCredentials(java.lang.String, java.lang.String,
+     * java.lang.String)
+     */
+    @Override
+    public Credentials getCredentials(final String password, final String keyLabel, final String certLabel) {
+
+        try {
+            // Obtain the Luna Keystore - Access the LunaSA via PKCS11 through
+            // the Luna Provider
+            final KeyStore lunaKeyStore = KeyStore.getInstance("Luna");
+            lunaKeyStore.load(null, null); // Can be null-null after login
+
+            // List the LunaSA contents
+            LOGGER.info("Luna Keystore contains");
+            final Enumeration<String> aliases = lunaKeyStore.aliases();
+            while (aliases.hasMoreElements()) {
+                final String keyStoreObj = aliases.nextElement();
+                if (LOGGER.isLoggable(Level.INFO)) {
+                    LOGGER.info("\t-" + keyStoreObj);
+                }
+            }
+
+            // Retrieve the PrivateKey and Certificate by labels
+            final PrivateKey privateKey = (PrivateKey) lunaKeyStore.getKey(keyLabel, password.toCharArray());
+            final X509Certificate cert = (X509Certificate) lunaKeyStore.getCertificate(certLabel);
+            final X509Certificate[] certChain = new X509Certificate[1];
+            certChain[0] = cert;
+
+            // Create credentials
+            final CredentialFactory credentialFactory = CredentialFactory.newInstance();
+            final PrivateKeyHolder pkh = PrivateKeyHolderFactory.newInstance().createPrivateKey(privateKey,
+                                                                                                "LunaProvider");
+            return credentialFactory.createCredentials(pkh, certChain[0], certChain);
+        } catch (KeyStoreException | NoSuchAlgorithmException | CertificateException
+                 | IOException | UnrecoverableKeyException
+                 | PDFInvalidParameterException e) {
+            if (LOGGER.isLoggable(Level.WARNING)) {
+                LOGGER.warning("Exception while obtaining LunaSA Credentials: " + e.getMessage());
+            }
```

Comment:
We should do more than log and return null for these exceptions. At minimum, we should throw them out to the client; we might even want to wrap them in a more general exception. My first pass would be: wrap all the Java security exceptions, and throw the `PDFInvalidParameterException` separately.

And yes, all of this will change the interface.